test/utility/misc.c

```diff
@@ -138,6 +138,44 @@ testutil_clean_work_dir(const char *dir)
     free(buf);
 }
 
+/*
+ * testutil_deduce_build_dir --
+ *     Deduce the build directory.
+ */
+void
+testutil_deduce_build_dir(TEST_OPTS *opts)
+{
+    struct stat stats;
+
+    char path[512], pwd[512], token_path[512], stat_path[512];
+    char *token;
+    size_t len;
+
+    token_path[0] = '\0';
+
+    if (getcwd(pwd, sizeof(pwd)) == NULL)
+        testutil_die(ENOENT, "No such directory");
+
+    testutil_check(__wt_snprintf(path, sizeof(path), "%s/%s", pwd, opts->argv0));
+
+    token = strtok(path, "/");
+
+    while (token != NULL) {
+        len = strlen(token);
+        strncat(token_path, "/", len);
+        strncat(token_path, token, len);
+
+        testutil_check(__wt_snprintf(stat_path, sizeof(stat_path), "%s/wt", token_path));
+
+        if (stat(stat_path, &stats) == 0) {
+            opts->build_dir = dstrdup(token_path);
+            return;
+        }
+        token = strtok(NULL, "/");
+    }
+    return;
```

Comment:
This approach is a top-down approach (if you draw your directory trees with the root directory on top).  If the executable is in "`/foo/bar/build/bench/tiered/push_pull"`, we'll look for /wt first, then /foo/wt, etc.  However, if you are on my machine, the executable is in `"/home/dda/wt/git/some-branch/build/bench/tiered/push_pull".`  And it will think the top of the build directory is `"/home/dda".`

To make that work (or if anybody puts their work in, say directory `"/wt"`, it's the same problem), I'd recommend a bottom up approach.  Start with path assigned to "/home/dda/wt/git/some-branch/build/bench/tiered/push_pull", and go in a loop until path is zero length.  Use strrchr(path, '/') to find the last '/' and put a '\0' there.  (exit loop if there is no '/'). Then you'll have path = "/home/dda/wt/git/some-branch/build/bench/tiered" and you can test if `%s/wt` exists. If yes, you're done, if not, path is already set up for the next iteration of the loop.

Reply:
Thanks for the suggestion. Updated the code to use the bottom-up approach to find the build dir.